rules/otbn.bzl

```diff
@@ -5,6 +5,36 @@
 load("//rules:opentitan.bzl", "OPENTITAN_PLATFORM", "opentitan_transition")
 load("//rules:bugfix.bzl", "find_cc_toolchain")
 
+def _otbn_assemble_sources(ctx):
+    """Helper function that, for each source file in the provided context, adds
+    an action to the context that invokes the otbn assember
+    (//hw/ip/otbn/util/otbn-as), producing a corresponding object file. Returns
+    a list of all object files that will be generated by these actions."""
+    objs = []
+
+    for src in ctx.files.srcs:
+      obj = ctx.actions.declare_file(src.basename.replace("." + src.extension, ".o"))
+      objs.append(obj)
+
+      ctx.actions.run(
+          outputs = [obj],
+          inputs = ([src] + ctx.files._otbn_as),
+          arguments = ["-o", obj.path, src.path],
+          executable = ctx.file._otbn_as,
+      )
```

Comment:
Alternatively, feel free to add an argument to the `otbn-as` and `otbn-ld` scripts in order to pass it explicitly rather than through the environment. That might make reproducing things a bit easier (I really hate environment variables because they're so hard to copy and paste!)

Reply:
Fixed it with environment variables for now, but an override might be a good thing to consider if we run into any similar issues in the future. The build was working for me before because `find_toolchain.py` looks in the default install location `/tools/riscv` if the environment variable isn't set, and I do indeed have the toolchain installed there.